lib/action_view/component/content_areas.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentAreas
+      extend ActiveSupport::Concern
+
+      included do
+        class_attribute :content_areas, default: []
+        self.content_areas = [] # default doesn't work until Rails 5.2
+      end
+
+      class_methods do
+        def set_content_areas(*areas)
+          attr_reader *areas
+          self.content_areas = areas.push(:body)
```

Comment:
Is there a reason that we always need a body to be present? If possible, I think it would be fairer to give users of `set_content_areas` the freedom to choose which types of content is needed for the component, rather than assuming that every component needs a body.

Reply:
fixed with undoing the content->body rename